BlueprintUICommonControls/Sources/AccessibilityLargeContentViewer.swift

```diff
@@ -0,0 +1,242 @@
+
+        var wrapping: Element
+
+        var configuration: LargeContentViewerConfiguration
+
+        public var content: ElementContent {
+            ElementContent(child: wrapping)
+        }
+
+        public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+            LargeContentViewerView.describe { config in
+                config[\.largeContentViewerConfiguration] = configuration
+            }
+        }
+    }
+}
+
+extension Accessibility {
+
+    private final class LargeContentViewerView: UIView, LargeContentViewerItem {
+
+        var largeContentViewerConfiguration: LargeContentViewerConfiguration {
+            didSet {
+                updateLargeContentViewerItem()
+            }
+        }
+
+        override init(frame: CGRect) {
+            largeContentViewerConfiguration = .init(display: .none)
+            super.init(frame: frame)
+            showsLargeContentViewer = false
+            updateLargeContentViewerItem()
+        }
+
+        required init?(coder: NSCoder) {
+            largeContentViewerConfiguration = .init(display: .none)
+            super.init(coder: coder)
+            showsLargeContentViewer = false
+            updateLargeContentViewerItem()
+        }
+
+        private func updateLargeContentViewerItem() {
+            scalesLargeContentImage = largeContentViewerConfiguration.scalesLargeContentImage
+            largeContentImageInsets = largeContentViewerConfiguration.largeContentImageInsets
+
+            switch largeContentViewerConfiguration.display {
+            case .title(let title, let image):
+                showsLargeContentViewer = true
+                largeContentTitle = title
+                largeContentImage = image
+            case .image(let image):
+                showsLargeContentViewer = true
+                largeContentTitle = nil
+                largeContentImage = image
+            case .none:
+                showsLargeContentViewer = false
+                largeContentTitle = nil
+                largeContentImage = nil
+            }
+        }
+
+        func didEndInteraction(at location: CGPoint, root: UICoordinateSpace) {
+            largeContentViewerConfiguration.interactionEndedCallback?(convert(location, from: root))
+        }
+    }
+}
+
+// MARK: - Large content viewer container
+
+extension Element {
+
+    /// Adds a large content viewer interaction container to the element.
+    /// This is used to wrap elements that need to be able to show a large content viewer.
+    /// Use this in conjunction with accessibilityShowsLargeContentViewer() on elements that need to show a large content viewer.
+    /// Elements that are wrapped in this container will be able to show a large content viewer and allow a user to swipe through them with one finger
+    /// and have the HUD update in real time.
+    public func accessibilityLargeContentViewerInteractionContainer() -> Element {
```

Comment:
Was just thinking about this this morning, and came to suggest this same thing, nice!